run_unit_tests.sh

```diff
@@ -52,15 +58,21 @@ if [ "$RUN_TESTS" == "1" ] ; then
         { bazel coverage --instrumentation_filter="-src/test" --combined_report=lcov \
             ${SHARED_OPTIONS} ${TEST_FILTER} \
             //src:ovms_test ${BAZEL_OPTIONS} > ${TEST_LOG} 2>&1 || \
-            test_fail_procedure; } && \
+            compress_logs && exit 1; } && \
             generate_coverage_report;
     fi
-    { 
-        bazel test --jobs=$JOBS ${BAZEL_OPTIONS} ${SHARED_OPTIONS} "${TEST_FILTER}" //src/python/binding:test_python_binding && \
-        bazel test \
-          ${SHARED_OPTIONS} "${TEST_FILTER}" \
-          //src:ovms_test ${BAZEL_OPTIONS} > ${TEST_LOG} 2>&1 || \
-          test_fail_procedure; } && \
-          test_success_procedure && \
-          rm -rf ${TEST_LOG};
+    bazel build ${SHARED_OPTIONS} "${TEST_FILTER}" //src:ovms_test ${BAZEL_OPTIONS}
+    set +x
+    failed=0
+    for i in `./bazel-bin/src/ovms_test --gtest_list_tests --gtest_filter="-LLMChatTemplateTest.*:LLMOptionsHttpTest.*" | grep -vE '^ ' | cut -d. -f1` ; do
+        ./bazel-bin/src/ovms_test --gtest_filter="$i.*" > tmp${TEST_LOG} 2>&1 || ( failed=1 ; echo $i ; cat tmp${TEST_LOG} ) 
+        cat tmp${TEST_LOG} >> ${TEST_LOG}
+    done
+    for i in `./bazel-bin/src/ovms_test --gtest_list_tests --gtest_filter="LLMChatTemplateTest.*:LLMOptionsHttpTest.*" | grep -v '^  '` ; do
+        ./bazel-bin/src/ovms_test --gtest_filter="*.$i" > tmp${TEST_LOG} 2>&1 || ( failed=1 ; echo "TEST NAME $i" ; cat tmp${TEST_LOG} ) 
+        cat tmp${TEST_LOG} >> ${TEST_LOG}
+    done    
+    grep -a " ms \| ms)" ${TEST_LOG}
```

Comment:
Do we have the same test count here after those changes?

Reply:
yes